core/src/main/java/org/apache/carbondata/core/datastore/blocklet/EncodedBlocklet.java

```diff
@@ -110,21 +120,24 @@ private void addEncodedMeasurePage(EncodedTablePage encodedTablePage) {
    * @param encodedTablePage
    * encoded table page
    */
-  private void addEncodedDimensionPage(EncodedTablePage encodedTablePage) {
+  private void addEncodedDimensionPage(EncodedTablePage encodedTablePage,
+      Map<String, LocalDictionaryGenerator> localDictionaryGeneratorMap) {
```

Comment:
Local dictionary map instance will not change and EncodedBlocklet instance is created only once, better to pass in constructor